### PR TITLE
feat: creating a starter workflow for browser testing

### PR DESCRIPTION
**Pull request by erunion**
| 🚥 Fix https://github.com/readmeio/oas-to-har/pull/78#discussion_r856377247 |
| :-- |

## 🧰 Changes

This copies across the Mocha + Chai + Karma browser testing workflow I've been building in `fetch-har` and `oas-to-har` into here so we can easily carry it to other projects.

https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization
https://github.com/readmeio/oas-to-har/pull/78

### CODE REVIEW

File: .github/workflow-templates/browser-testing.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          - windows-latest
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Install Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
```
Comment (kanadgupta) [inline .github/workflow-templates/browser-testing.yml]: Thoughts on just using Node v16 so we don't have to reinstall `npm`?

File: .github/workflow-templates/browser-testing.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Browser Testing
```
Comment (kanadgupta) [inline .github/workflow-templates/browser-testing.yml:1]: Wanna add some comments here about what this does and what config is required to run this? Like mocha config, the npm scripts required, etc.

**Review by kanadgupta — APPROVED** [review]

File: .github/workflow-templates/browser-testing.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        browser:
+          - chrome
+          - firefox
+        os:
+          - ubuntu-latest
+          - windows-latest
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
```
Comment (kanadgupta) [inline .github/workflow-templates/browser-testing.yml]: Curious if dependabot will pick up on workflows in this directory 🤔 
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflow-templates/browser-testing.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+
+      - name: Get npm cache directory
+        id: npm-cache-dir
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+
+      - uses: actions/cache@v2
```
Comment (kanadgupta) [inline .github/workflow-templates/browser-testing.yml]: This may be breaking, not sure
```suggestion
      - uses: actions/cache@v3
```

**Review by kanadgupta — CHANGES_REQUESTED** [review]
sorry, noticed a few outdated actions

File: .github/workflow-templates/browser-testing.properties.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Browser Testing",
+  "description": "Mocha + Chai + Karma browser testing starter workflow. To use this workflow you will need to have `chai`, `mocha`, `@jsdevtools/host-environment`, and `@jsdevtools/karma-config` installed, along with a `karma.config.js` file set up. Check out the [Karma config in `fetch-har`](https://github.com/readmeio/fetch-har/blob/main/karma.conf.js) for an example of how to set this up.",
```
Comment (kanadgupta) [inline .github/workflow-templates/browser-testing.properties.json:3]: I'm curious about where this description will show up when we try and set up this workflow 🤔  I was thinking we include this in a comment block in the workflow file itself so devs will see it when writing their workflow file? I think this is good enough for now though.

**Review by kanadgupta — APPROVED** [review]

File: .github/workflow-templates/browser-testing.properties.json (hunk 5 above)
Comment (erunion) [reply .github/workflow-templates/browser-testing.properties.json:3]: Would rather it be in a README or something but I'm not actually sure how this is surfaced. Will see after I merge this in and tweak as needed.

**Pull request merged**